pkg/runtime/wfengine/wfengine_backend_test.go

```diff
@@ -0,0 +1,119 @@
+
+	"github.com/dapr/dapr/pkg/config"
+	"github.com/dapr/dapr/pkg/runtime/wfengine"
+	daprt "github.com/dapr/dapr/pkg/testing"
+	"github.com/dapr/kit/logger"
+)
+
+const SqliteBackendType = "workflowbackend.sqlite"
+
+func init() {
+	wfengine.SetLogLevel(logger.DebugLevel)
+}
+
+func TestSqliteBackendSetupWithMetadata(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockSqliteBackendManager)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
+
+	_, ok := engine.Backend.(*wfengine.ActorBackend)
+
+	assert.False(t, ok, "actor is not of type ActorBackend")
+	assert.Equal(t, SqliteBackendType, engine.BackendType)
+}
+
+func TestSqliteBackendSetupWithoutMetadata(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockSqliteBackendManagerWithoutMetadata)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
+
+	_, ok := engine.Backend.(*wfengine.ActorBackend)
+
+	assert.False(t, ok, "actor is not of type ActorBackend")
+	assert.Equal(t, SqliteBackendType, engine.BackendType)
+}
+
+func TestActorBackendSetup(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockActorBackendManager)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
+
+	_, ok := engine.Backend.(*wfengine.ActorBackend)
+
+	assert.True(t, ok, "actor is of type ActorBackend")
+	assert.Equal(t, wfengine.ActorBackendType, engine.BackendType)
+}
+
+func TestInvalidBackendSetup(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockInvalidaBackendManager)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
+
+	assert.Nil(t, engine.Backend, "Backend is set to nil")
+}
+
+func TestBackendComponentNilSetup(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockNilBackendComponentManager)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
+
+	_, ok := engine.Backend.(*wfengine.ActorBackend)
+
+	assert.True(t, ok, "actor is of type ActorBackend")
+	assert.Equal(t, wfengine.ActorBackendType, engine.BackendType)
+}
+
+func TestInvalidSqliteBackendComponentSetup(t *testing.T) {
+	mockWorkflowBackendManager := new(daprt.MockSqliteBackendComponentInvalidManager)
+	spec := config.WorkflowSpec{MaxConcurrentWorkflowInvocations: 100, MaxConcurrentActivityInvocations: 100}
+
+	engine := wfengine.NewWorkflowEngine(testAppID, spec, mockWorkflowBackendManager)
```

Comment:
if a component is configured in an invalid way, would it be better to fail the engine startup rather than quietly defaulting to the actor backend?

Reply:
Agree

Reply:
If component is misconfigured, we don't fall back to actor, we are [logging the error message](https://github.com/dapr/dapr/pull/7283/files#diff-500c4c9a401c87ceef033c9e69aff0b306bd732e68e521ca8bd9892cb50bb7a4R83-R85) and backend is not initialized.

Are you suggesting to not set the [`a.workflowEngine = wfe`](https://github.com/dapr/dapr/pull/7283/files#diff-e56f9bb947923c6f8bb4c995bf4b9469a0ed117276223d16b177dcb04399a4bcR462-R464) if backend is not configured.